templates/etc/docker/daemon.json.j2

```diff
@@ -67,4 +67,7 @@
 {%-   endfor %}
 {%    set _ = docker__tpl_options.update({"storage-opts": docker__tpl_storage_options}) %}
 {%- endif %}
+{%  if docker__arb_daemon_opts|d() -%}
+{%    set _ = docker__tpl_options.update(docker__arb_daemon_opts|from_yaml) %}
```

Comment:
IMHO the use of the `from_yaml` Jinja2 filter is redundant/misused here. Ansible does already converted `docker__arb_daemon_opts` to a Python dict at this point.